cfgov/preprocessed/js/modules/util/EventObserver.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+/**
+ * EventObserver
+ * @class
+ *
+ * @classdesc Used for creating an object
+ *   that can be used to dispatch and listen to custom events.
+ * @returns {Object} An EventObserver instance.
+ */
+function EventObserver() {
+
```

Comment:
After working with this I don't think the native Event constructor is a fit. The custom events on the MDN URL are DOM elements. `EventObserver` is for adding event dispatching methods to a non-DOM class instance that work in a way familiar to the DOM methods. It would be nice to not have to pass a `target` property to `dispatchEvent`, but I'm not seeing a way with custom events. Let me know if you see way!